runtime/near-vm-logic/src/logic.rs

```diff
@@ -628,6 +630,28 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.block_index)
     }
 
+    /// Returns the current block hash.
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    #[cfg(feature = "protocol_feature_vm_hash")]
+    pub fn block_hash(&mut self) -> Result<BlockHash> {
```

Comment:
I'm a fan of full type path here so that you don't need the feature check at the top of the file.  But again it's a personal preference.

Reply:
If we want to have block hashes - let's surface all the available block hashes - e.g. 2 last epochs.

Reply:
Good call

Reply:
@ilblackdragon You are right, I do need to add in the method to get a block from an index. Ach.

Reply:
Please sync with @bowenwang1996 to make sure we are not attempting to surface more block hashes than what is available with GC.

I also don't think surfacing more than let's say 10 block hashes is safe. We need to be super conservative with what we are exposing to contracts, because it cannot be undone. If we surface 2 epochs worth of hashes, then we cannot ever change the length of the epoch after that because some contracts may start breaking. And we did change the length of the epoch in past. We also do not want to surface number which is too large, because who knows how aggressive we would want our GC to be in the future. But surfacing something like 10 hashes seems arbitrary. Therefore I suggest we do not surface more than one or two hashes.

Reply:
@nearmax I understand the concern. However, if we can't get at the last 256 hashes (needed by the [EVM's `BLOCKHASH` instruction](http://ethervm.io/)), then we will not be able to go with the scheme you outlined in https://github.com/near/nearcore/issues/3456#issuecomment-793048499, but will have to compute the EVM's block hashes based on the NEAR block number instead of the NEAR block hash. Sounds like that'll be the safer thing to do, despite the temporary fork concern you raised there.

Reply:
I think exposing last 256 blocks should not be a problem, although it requires more effort in the implementation to make sure that it is not expensive. @artob is there any alternative solution to exposing the last 256 block hashes?